Source/ACE/Entity/Vendor.cs

```diff
@@ -58,34 +36,139 @@ public override void OnUse(ObjectGuid playerId)
+        /// </summary>
+        private void LoadInventory()
+        {
+            // Load Vendor Inventory from database.
+            if (!inventoryloaded)
+            {
+                List<VendorItems> items = new List<VendorItems>();
+                items = DatabaseManager.World.GetVendorWeenieInventoryById(AceObject.WeenieClassId);
+                foreach (VendorItems item in items)
                 {
-                    serverMessage = "Your way to far away to trust, come closer and buy something or leave me alone!";
+                    WorldObject wo = WorldObjectFactory.CreateNewWorldObject(item.WeenieClassId);
+                    if (wo != null)
+                    {
+                        wo.ContainerId = Guid.Full;
+                        defaultItemsForSale.Add(wo.Guid, wo);
+                    }
                 }
-                else
+                inventoryloaded = true;
+            }
+        }
+
+        /// <summary>
+        /// Buys Items from player
+        /// </summary>
+        /// <param name="vendorid">GUID of Vendor</param>
+        /// <param name="items">Item Profile, Ammount and ID</param>
+        /// <param name="player"></param>
+        public void BuyItems(ObjectGuid vendorid, List<ItemProfile> items, Player player)
+        {
+            // do you have enough cash / iventory space for all this shit.
+            int goldcost = 0;
+            List<WorldObject> purchaselist = new List<WorldObject>();
+
+            // que transactions.
+            foreach (ItemProfile item in items)
+            {
+                // check default items for id
+                if (defaultItemsForSale.ContainsKey(item.Guid))
                 {
-                    // create the outbound server message
-                    serverMessage = "You Break it, you bought it!";
+                    // todo: more stack logic ?
+                    while (item.Amount > 0)
+                    {
+                        WorldObject wo = WorldObjectFactory.CreateNewWorldObject(defaultItemsForSale[item.Guid].WeenieClassId);
+                        // can we stack this ?
+                        if (wo.MaxStackSize.HasValue)
+                        {
+                            if ((wo.MaxStackSize.Value != 0) & (wo.MaxStackSize.Value <= item.Amount))
+                            {
+                                item.Amount -= wo.MaxStackSize.Value;
+                                goldcost += (int)defaultItemsForSale[item.Guid].Value.Value * wo.MaxStackSize.Value;
+                                wo.StackSize = wo.MaxStackSize.Value;
+                                purchaselist.Add(wo);
+                            }
+                            // else we cant stack it or its less then max stack size.
+                            else
+                            {
+                                item.Amount -= item.Amount;
```

Comment:
this looks... wrong.  just assign it to 0?  not sure what this is supposed to be.